src/cljs/harja/views/hallinta/kojelauta.cljs

```diff
@@ -0,0 +1,131 @@
+                             1 (:nimi (first %))
+                             (str (count %) " urakkaa valittu"))}
+     (r/wrap (:urakat valinnat) #(e! (tiedot/->Valitse :urakat %)))]]])
+
+
+(defn kustannussuunitelman-tila-sarake
+  [rivi]
+  (let [{:keys [vahvistamattomia vahvistettuja suunnitelman_tila]} (:ks_tila rivi)]
+    [yleiset/wrap-if true
+     [yleiset/tooltip {} :% "Klikkaa siirtyäksesi urakan kustannussuunnitelmaan"]
+     [:div.klikattava {:on-click #(do
+                                    (prn "Jarno on click! urakassa " (:id rivi) " ja elyssä " (:ely_id rivi))
+                                    (siirtymat/kustannusten-seurantaan-valitussa-urakassa (:ely_id rivi) (:id rivi)))}
+      (cond
+        (= "aloittamatta" suunnitelman_tila)
+        (yleiset/tila-indikaattori "hylatty" {:fmt-fn (constantly "Aloittamatta")})
+
+        (= "aloitettu" suunnitelman_tila)
+        (yleiset/tila-indikaattori "kesken" {:fmt-fn #(str "Aloitettuja: " vahvistamattomia
+                                                        ", vahvistettuja: " vahvistettuja)})
+
+        (= "vahvistettu" suunnitelman_tila)
+        (yleiset/tila-indikaattori "valmis" {:fmt-fn (constantly "Valmis")}))]]))
+
+(defn listaus [e! {:keys [valinnat urakat urakoiden-tilat] :as app}]
+  (let [valitut-urakat (:urakat valinnat)
+        valittu-ely (get-in valinnat [:ely :id])
+        valittu-hk-alkuvuosi (:urakkavuosi valinnat)
+        ;; ely-suodatus
+        urakat (if (nil? valittu-ely)
+                 urakat
+                 (filter #(= valittu-ely (:ely_id %)) urakat))
+        ;; hoitokausisuodatus valittu-hk-alkuvuosi
+        urakat (if (nil? valittu-hk-alkuvuosi)
+                 urakat
+                 (filter #(= valittu-hk-alkuvuosi (:hoitokauden_alkuvuosi %)) urakat))
+        ;; urakkasuodatus
+        urakat (if (empty? valitut-urakat)
+                 urakat
+                 (filter #((into #{} (map :id valitut-urakat)) (:id %)) urakat))]
+    [:div
+     [debug/debug urakat]
+     [grid/grid
+      {:otsikko (str "Urakoiden tilat")
+       :tyhja (if (nil? urakat) [ajax-loader "Ladataan tietoja"] "Ei tietoja, tarkistathan valitut suodattimet.")}
+      [{:otsikko "Urakka"
+        :tyyppi :string
+        :nimi :nimi
+        :leveys 5
+        :muokattava? (constantly false)}
+       {:otsikko "Hoito\u00ADvuosi"
+        :muokattava? (constantly false)
+        :nimi :hoitokauden_alkuvuosi :leveys 3
+        :tyyppi :string :fmt #(pvm/hoitokausi-str-alkuvuodesta %)}
+       {:otsikko "Kustannus\u00ADsuunnitelma"
+        :muokattava? (constantly false)
+        :nimi :ks_tila :leveys 15
+        :tyyppi :komponentti
+        :komponentti (fn [rivi] [kustannussuunitelman-tila-sarake rivi])}]
+      urakat]]))
+
+
+(defn kojelauta* [e! app]
+  (komp/luo
+    (komp/sisaan #(e! (tiedot/->HaeUrakat)))
+    (fn [e! app]
+      [:div.kojelauta-hallinta
+       [:h1 "Etusivu"]
+       [suodattimet e! app]
+       [debug/debug app]
```

Comment:
Poista kaikki nämä debug komponentit, tai lisää iffi jos ollaan kehitysympäristössä, kun nämä on nyt menossa tuotantoon asti. Näitä on 2 kappaletta.